projects/sandbox/train/train/utils.py

```diff
@@ -0,0 +1,135 @@
+from collections import OrderedDict
+from math import pi
+from pathlib import Path
+from typing import Optional, Tuple, TypeVar
+
+import h5py
+import numpy as np
+import torch
+
+from bbhnet.data.distributions import Cosine, LogNormal, Uniform
+from bbhnet.data.glitch_sampler import GlitchSampler
+from ml4gw.transforms import RandomWaveformInjection
+
+Tensor = TypeVar("T", np.ndarray, torch.Tensor)
+
+
+class MultiInputSequential(torch.nn.Sequential):
+    """
+    Cheap wrapper around the torch Sequential object
+    to support multiple input
+    """
+
+    def forward(self, *inputs):
+        for module in self._modules.values():
+            if isinstance(inputs, tuple):
+                inputs = module(*inputs)
+            else:
+                inputs = module(inputs)
+        return inputs

```

Comment:
Is this to support augmenting the background with glitches and signals, as well as changing target tensors

Reply:
Yeah exactly. Part of me thinks it might make more sense to make an `Augmenter` module the same way we have a `Preprocessor` module and handle this more explicitly there (i.e. have the injector and inserter as attribute submodules of the `Augmenter` submodule and have the forward method look like
```python
def forward(self, X, y):
    X = self.inserter(X)
    X, y = self.injector(X, y)
    return X, y
```
but there's just a couple implementation details about how/where it would get initialized that I don't really want to bog this PR down with.